### ISSUE TITLE
improvements on current recipe format

### ISSUE THREAD

**Issue by semio** [semio/ddf_utils#85]
These days I spent a few hours implementing the syntax I suggested in #57, and then I figured out there are some thing we can improve on the yaml recipe.

When I was writing a long recipe, I felt it easy to lost the overview of the recipe. Even when I fold all levels in the recipe, I still can't get the overview of it:

<img width="531" alt="2017-11-24 15 44 23" src="https://user-images.githubusercontent.com/351856/33212358-a19b2544-d15d-11e7-8918-2da008f6a84e.png">

All I can see is just procedure names. To resolve this, we might rearrange the keyword order in recipe, to list the result id at first:

<img width="522" alt="2017-11-24 16 07 45" src="https://user-images.githubusercontent.com/351856/33212457-339404ca-d15e-11e7-962f-80ddb8669754.png">

But there is still problem. In a long recipe, such as procedures to generate population-age_group from population by age gender, for each group we need at least 3 procedures: filter target ages, groupby and aggregate, translate the indicator name. When there are 10+ age groups to do, we will see a long list of ingredient results, and not easy to see which ones are we interested in. So to improve this, I suggest adding one more level to list the ingredients we are interested in: 

<img width="402" alt="2017-11-24 21 43 00" src="https://user-images.githubusercontent.com/351856/33213030-a4dbee98-d160-11e7-9170-ced6ecb3cfbe.png">

**Comment by jheeffer** [comment]
First thing that comes to mind is how to still have template strings for datapoints? Or have multiple datapoints in one (final) ingredient?
E.g. for Systema Globalis, will you have 500 datapoints chains?

**Comment by jheeffer** [comment]
Chaining in itself is not a bad thing. We do still need branching. Like #59 maybe. I guess that's what your proposing. So not 500 chains for SG. I'll look at it more in the morning.

**Comment by semio** [comment]
yes, the block names are ingredient id for the output of last procedure in the block. Then we can use them in another block.
